torch/_dynamo/variables/base.py

```diff
@@ -256,6 +256,7 @@ def __init__(
         self,
         guards: Optional[Set] = None,
         source: Source = None,
+        dynamic_spec = None,
```

Comment:
I don't think the dynamic_spec should go on the base class for VariableTracker, it seems specific to Tensors.  

Could we encode this in the shape env?